wallet/src/client/actor.rs

```diff
@@ -0,0 +1,96 @@
+//! JsonRPC client that communicates the wallet with the node.
+
+use actix::prelude::*;
+use async_jsonrpc_client::{
+    transports::{shared::EventLoopHandle, tcp::TcpSocket},
+    Transport,
+};
+use futures::Future;
+pub use jsonrpc_core as rpc;
+
+/// JsonRPC client.
+#[derive(Debug)]
+pub struct JsonRpc {
+    handle: EventLoopHandle,
+    s: TcpSocket,
+    // subscriptions: Subscriptions,
+    // node_id_to_client_id: HashMap<SubscriptionId, SubscriptionId>,
+}
+
+impl JsonRpc {
+    fn new(url: &str) -> Self {
+        let (handle, s) = TcpSocket::new(url).unwrap();
+        Self {
+            handle,
+            s,
+            // subscriptions: Default::default(),
+            // node_id_to_client_id: Default::default(),
+        }
+    }
+}
+
+impl Default for JsonRpc {
+    fn default() -> Self {
+        Self::new("127.0.0.1:1234")
+    }
+}
+
+impl Actor for JsonRpc {
+    type Context = Context<Self>;
+}
+
+impl Supervised for JsonRpc {}
+
+impl SystemService for JsonRpc {}
+
+/// A JsonRPC request.
+pub struct Request {
+    pub method: String,
+    pub params: Option<rpc::Value>,
```

Comment:
This could be a `jsonrpc::Params` initialized to `None`.
```suggestion
    pub params: rpc::Params,
```

Then, the params method would be just

```rust
self.params = params;
self
```

Reply:
👌nice one